File: CLI/GameClasses/AccessoryEngineData.cs
using System.Numerics;
using SiiParser;
using SiiParser.Attributes;

namespace SiiParserSCS.GameClasses
{
    [SiiUnit("accessory_engine_data")]
    internal class AccessoryEngineData : GameClass
    {
        [SiiAttribute("info")]
        public string[] Info { get; set; }
        
        [SiiAttribute("torque")]
        public float Torque { get; set; }
        
        [SiiAttribute("torque_curve")]
        public Vector2 TorqueCurve { get; set; }
        
        [SiiAttribute("rpm_idle")]
        public float RpmIdle { get; set; }
        
        [SiiAttribute("rpm_limit")]
        public float RpmLimit { get; set; }
        
        [SiiAttribute("adblue_consumption")]
        public float AdblueConsumption { get; set; }
    }
}

File: src/game-classes/accessory-engine-data.ts
import SiiUnit from "../decorators/sii-unit";
import SiiAttribute from "../decorators/sii-attribute";
import GameClass from "./game-class";
import Vector2 from "../classes/vector-2";

@SiiUnit("accessory_engine_data")
class AccessoryEngineData extends GameClass {
    namespace: string;

    @SiiAttribute("info[]")
    public info: string[];

    @SiiAttribute("torque")
    public torque: number;

    @SiiAttribute("torque_curve[]")
    public torqueCurve: Vector2[];

    @SiiAttribute("rpm_idle")
    public rpmIdle: number;

    @SiiAttribute("rpm_limit")
    public rpmLimit: number;

    @SiiAttribute("adblue_consumption")
    public adblue_consumption: number;
}

export default AccessoryEngineData;